services/shippable/shippable.service.js

```diff
@@ -1,23 +1,52 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { checkErrorResponse } = require('../../lib/error-helper')
+const BaseJsonService = require('../base-json')
+const Joi = require('joi')
+const { InvalidResponse, NotFound } = require('../errors')
+const { nonNegativeInteger } = require('../validators')
+
+// source: https://github.com/badges/shields/pull/1362#discussion_r161693830
+const statusCodes = {
+  0: { color: '#5183A0', label: 'waiting' },
+  10: { color: '#5183A0', label: 'queued' },
+  20: { color: '#5183A0', label: 'processing' },
+  30: { color: '#44CC11', label: 'success' },
+  40: { color: '#F8A97D', label: 'skipped' },
+  50: { color: '#CEA61B', label: 'unstable' },
+  60: { color: '#555555', label: 'timeout' },
+  70: { color: '#6BAFBD', label: 'cancelled' },
+  80: { color: '#DC5F59', label: 'failed' },
+  90: { color: '#555555', label: 'stopped' },
+}
+
+const schema = Joi.array()
+  .items(
+    Joi.object({
+      branchName: Joi.string().required(),
+      statusCode: nonNegativeInteger,
```

Comment:
Can status code be any non-negative integer, or will it always be one of the values from the above set (0, 10, 20, ... 90)?